Tools/LambdaTestTool/src/Amazon.Lambda.TestTool.BlazorTester/Controllers/InvokeApiController.cs

```diff
@@ -0,0 +1,239 @@
+        public async Task<IActionResult> ExecuteFunction()
+        {
+            if (!TryGetConfigFile(out var lambdaConfig))
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError,
+                    new InternalException("ServiceException", "Error while loading function configuration"));
+            }
+
+            if (lambdaConfig.FunctionInfos.Count == 0)
+            {
+                return NotFound(new InternalException("ResourceNotFoundException", "Default function not found"));
+            }
+
+            return Ok(await ExecuteFunctionInternal(lambdaConfig, lambdaConfig.FunctionInfos[0]));
+        }
+
+        [HttpPost("execute/{functionName}")]
+        [HttpPost("2015-03-31/functions/{functionName}/invocations")]
+        public async Task<object> ExecuteFunction(string functionName)
+        {
+            if (!TryGetConfigFile(out var lambdaConfig))
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError,
+                    new InternalException("ServiceException", "Error while loading function configuration"));
+            }
+
+            var functionInfo = lambdaConfig.FunctionInfos.FirstOrDefault(f => f.Name == functionName);
+            if (functionInfo == null)
+            {
+                return NotFound(new InternalException("ResourceNotFoundException",
+                    $"Function not found: {functionName}"));
+            }
+
+            return Ok(await ExecuteFunctionInternal(lambdaConfig, functionInfo));
+        }
+
+        private bool TryGetConfigFile(out LambdaConfigInfo configInfo)
+        {
+            configInfo = null;
+
+            if (_lambdaConfig != null)
+            {
+                configInfo = _lambdaConfig;
+                return true;
+            }
+
+            if (_lambdaOptions.LambdaConfigFiles.Count == 0)
+            {
+                Console.Error.WriteLine("LambdaConfigFiles list is empty");
+                return false;
+            }
+
+            var configPath = _lambdaOptions.LambdaConfigFiles[0];
+            try
+            {
+                configInfo = LambdaDefaultsConfigFileParser.LoadFromFile(configPath);
+                _lambdaConfig = configInfo;
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine("Error loading lambda config from '{0}'", configPath);
+                Console.Error.WriteLine(e.ToString());
+            }
+
+            return true;
+        }
+
+        private async Task<object> ExecuteFunctionInternal(LambdaConfigInfo lambdaConfig,
+            LambdaFunctionInfo functionInfo)
+        {
+            var requestReader = new LambdaRequestReader(Request);
+            var function = _lambdaOptions.LoadLambdaFuntion(lambdaConfig, functionInfo.Handler);
+
+            var request = new ExecutionRequest
+            {
+                Function = function,
+                AWSProfile = lambdaConfig.AWSProfile,
```

Comment:
We need some way on either a per invoke or per session users can configure the profile and region. This feature won't work for users that don't want profiles and regions in their config file. I think it is fine using the config as default the. Also we might want to use the SDK's credential and region fallback chain if neither the config or a specific profile and region are not specified. Also can we add a log message which profile and region were used?

Reply:
In [LambdaExecutor](https://github.com/aws/aws-lambda-dotnet/blob/master/Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Runtime/LambdaExecutor.cs#L31) these config parameters are passed to env variables if they are set. If you don't fill it in the config file, you can pass these parameters directly via environment variables when you start debugging. What we actively use, for example, through `launchSettings.json`.

At what level do you want to see `profile` and `region` logging? I propose to show the values of the corresponding environment variables at `LambdaExecutor` level. Because these are the values that will already be used later in the execution of the function